llama-index-core/llama_index/core/postprocessor/optimizer.py

```diff
@@ -63,6 +67,9 @@ def __init__(
         )
         response = query_engine.query("<query_str>")
         """
+        embed_model = embed_model or embed_model_from_settings_or_context(
+            Settings, None
+        )
```

Comment:
Currently `embed_model_from_settings_or_context` marks the parameter `context` as optional but gives no default value for it:

https://github.com/run-llama/llama_index/blob/4c0a6172226214b92275e9e2c5eb0d21856d6863/llama-index-core/llama_index/core/settings.py#L267-L270

If we give it a default value (i.e. `None`), then there's no need for users to pass the second parameter `None`.

Reply:
If `service_context` has been deprecated, maybe here I should just use `Settings.embed_model` instead of `embed_model_from_settings_or_context(Settings, None)`?